ui/src/helper-functions.ts

```diff
@@ -0,0 +1,7 @@
+interface DateObject {
```

Comment:
Hey Brew! I noticed these date helpers from a (comparatively) old PR. Would you help me understand what they're doing? Date/time/zone handling is notoriously pernicious, so I want to make sure nothing scary is happening.

Reply:
Right! Yes, this is a really tricky thing. The answer is that something scary was happening, and this was the solution that we came to. 

So, basically there's a bug in swagger codegen that means that it doesn't actually properly deserialize dates. So the flow on dates is basically this:
The API call returns an object. The code expects it to be a `Date` object, but it is actually a generic object with date like properties (millis, monthOfYear, etc.) So while it had a lot of the properties of a Date, it didn't have any of the methods. However, the compiler expected it to be a Date object, so it didn't actually let me access the properties. This is basically a method of resetting that generic objects so the compiled code lines up with the typescript representation.